.ci/test_r_package_windows.ps1

```diff
@@ -28,6 +28,24 @@ function Run-R-Code-Redirect-Stderr {
   Rscript --vanilla -e $decorated_code
 }
 
+# Remove all items matching some pattern from PATH environment variable
+function Remove-From-Path {
+  param(
+    [string]$pattern_to_remove
+  )
+  $env:PATH = ($env:PATH.Split(';') | Where-Object { $_ -notmatch "$pattern_to_remove" }) -join ';'
+}
+
+# remove some details that exist in the GitHub Actions images which might
+# cause conflicts with R and other components installed by this script
+$env:RTOOLS40_HOME = ""
+Remove-From-Path ".*chocolatey.*"
+Remove-From-Path ".*Chocolatey.*"
+Remove-From-Path ".*Git.*mingw64.*"
+Remove-From-Path ".*msys64.*"
+Remove-From-Path ".*rtools40.*"
```

Comment:
WDYT about to remove this particular directory physically? 
Looks like there could be conflicts during our own installation 
https://github.com/microsoft/LightGBM/blob/3a5e3c001f7a571eec375d021d12ccd37353a60c/.ci/test_r_package_windows.ps1#L45

Reply:
I originally had something like `Remove-Item "C:\rtools40" -Recurse"`, but it was failing with an error saying something like "insufficient permission to remove this, must be an administrator or this may be a system file" (don't remember the exact error message).

Although, I was just looking at your work in RGF...maybe we could borrow your solution from there to get past those error? :grinning: 

https://github.com/RGF-team/rgf/blob/ab328a01d3af809c0c1fd4122d8b70b9fb2e9757/.ci/r_tests_windows.ps1#L20

Reply:
I think so! This command takes some time (several seconds) but I think it won't be excess and can protect us from some other problems.

Reply:
BTW, I don't know why, but I had to manually install `qpdf` for Windows R jobs (I chose `pacman` package manager because it's bundled with Rtools and one-line solution) to pass `--as-cran` checks for vignettes.
Maybe this info will help you to save some time on other projects or even here in LightGBM.
https://github.com/RGF-team/rgf/blob/ab328a01d3af809c0c1fd4122d8b70b9fb2e9757/.ci/r_tests_windows.ps1#L42

Reply:
thanks! Maybe we'll have to worry about `qpdf` on #3946 , actually, so that is timely advicce